temprl/wrapper.py

```diff
@@ -90,6 +90,30 @@ def step(self, symbol: Interpretation) -> Tuple[State, float]:
         """
         return self._simulator.step(symbol)
 
+    def current_dfa_state(self) -> State:
+        """Get the current DFA state."""
+        return self._simulator._current_state
+
+
+class StepController:
+    def __init__(self,
+                 step_func: Callable[[Set[str]], bool],
+                 allow_first: bool = True):
+        self.started = False
+        self.step_func = step_func
+        self.allow_first = allow_first
+
+    def check(self,
+              fluents) -> bool:
+        if self.allow_first and not self.started:
+            self.started = True
+            return True
+        may_step = self.step_func(fluents)
+        return self.started and may_step
```

Comment:
I see a problem here. Assume `allow_first` is False. The if condition at line `108` is never satisfied. Therefore, `self.started` is never set to `True`. Hence, the return value of `check` is always `False`, regardless of the value of `may_step`.

I think this is not the intended behaviour. `self.allow_first` should only be considered at the first step, i.e. when `self.started` is False. Do you agree? 


```suggestion
    def check(self,
              fluents) -> bool:
        if self.allow_first and not self.started:
            self.started = True
            return True
        may_step = self.step_func(fluents)
        result = self.started and may_step
        self.started = True
        return result
```

Reply:
With the suggested change, if `allow_first = False`, `result = False` only the first time, but then it will be equal to `result = may_step`.

Reply:
I don't think this would be correct either: if `self.allow_first` is set to `False`, then `self.started` should become `True` only the first time `self.step_func(fluents)` returns `True`. In the proposed suggestion, if `self.step_func(fluents)` evaluates to `True`, no step will take place as `result` will be `False`. I will push a new commit shortly with this quickfix.

Reply:
OK sorry, I thought `started` was tracking the underlying environment, and not the trace filtered by `step_func`. Looking forward to the fix.